### DISCUSSION TITLE
Can I check a Nu script without running it?

### DISCUSSION THREAD

**Discussion by rycee** [nushell/nushell#5702] [Q&A] [answered]
I'm wondering if there is a way that I can check that a file is a valid Nu script using the `nu` command? Something simple like checking for valid syntax would be sufficient for my purposes. Basically, I would like something similar to `bash -n file.sh`.

**Comment by fdncred** [comment, marked as answer]
I don't think we have such a thing for scripts. You can type on the prompt and see mistakes because syntax highlighting shows them. It would be good to have a `nu check` command.

**Reply by rycee** [reply]
Thanks for the reply! I'm experimenting a bit with using Nu as the scripting language for some generated code where Bash is currently used. Since the code is generated it is very helpful to have a sanity check of the output (without actually running it). I will keep an eye out for future support of a check command in `nu`.

**Comment by kubouch** [comment]
@rycee That would be an interesting addition: we could add a `nu check` command that would parse the file, but not run it, and it would also report errors. Can you file a feature request issue for it?

Right now, we have the `nu-highlight` command that parses the source code and syntax-highlight it. It might help for your use case.

**Reply by rycee** [reply]
Apologies for the long delay, I created https://github.com/nushell/nushell/issues/5825 just now.

**Reply by rycee** [reply]
Oh, and `nu-highlight` will probably suffice. I can just pipe the output to `/dev/null`.
